### ISSUE TITLE
Bundows socket connect huge delay (20 sec)

### ISSUE THREAD

**Issue by Ivan-Baranov** [oven-sh/bun#10918]
### What version of Bun is running?

1.1.7

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64

### What steps can reproduce the bug?

```js
// test.js

const { Socket } = require('node:net')
const h = new Socket()

console.time('S')
h.on('connect', () => console.timeLog('S', 'Connected'))
h.on('ready', () => console.timeLog('S', 'Ready'))
h.connect(6432, 'rc1b-zzz.mdb.yandexcloud.net')
```

### What is the expected behavior?

```
// node v20.12.2
// node test.js
S: 35.637ms Connected
S: 38.215ms Ready
```

### What do you see instead?

```
// Bun v1.1.7
// bun test.js
[21.06s] S Ready
[21.07s] S Connected
```

**Comment by Electroid** [comment]
This is probably because we need to support happy-eyeballs.

**Comment by Electroid** [comment]
Duplicate of #9658

**Comment by Jarred-Sumner** [comment]
The error is not propagating to the uncaught exception handler. It is failing silently.

This logs the error:
```js
const { Socket } = require('node:net')
const h = new Socket()

console.time('S')
h.on('connect', () => console.timeLog('S', 'Connected'))
h.on('ready', () => console.timeLog('S', 'Ready'))

h.connect(6432, 'rc1b-zzz.mdb.yandexcloud.net')
```

```js
❯ bun foo.js
ECONNREFUSED: Failed to connect
   errno: 8
 syscall: "connect"

```

**Comment by Ivan-Baranov** [comment]
Fixed in 1.1.9
